client/me/billing-history/receipt.jsx

```diff
@@ -58,9 +62,26 @@ class BillingReceipt extends React.Component {
 			transaction,
 			transactionFetchError,
 			transactionId,
+			translate,
 		} = this.props;
 
 		if ( transactionFetchError ) {
+			if ( 'invalid_receipt' === transactionFetchError.error ) {
+				this.props.errorNotice(
+					translate( 'Could not find Receipt #%s', { args: transactionId } ),
```

Comment:
Based on a [search for other error messages](https://github.com/Automattic/wp-calypso/search?q=errorNotice(&type=&utf8=%E2%9C%93), I think this message could be a little more conversational.

How about:
> Sorry, we couldn't find receipt #123.